### ISSUE TITLE
Create SMS delivery failure sign in flow in homepage

### ISSUE THREAD

**Issue by carlosmiceli** [Expensify/App#52086]
We need to replicate the flow that currently exists in staging.expensify.com when trying to sign in with a blacklisted phone number (see steps in screenshots). 

This is triggered when the user tries to sign in with a phone number (account must already be created with said number) and receives `hasSMSDeliveryFailure: true` when calling `BeginSignIn`:

<img width="627" alt="Screenshot 2024-11-05 at 11 34 38 PM" src="https://github.com/user-attachments/assets/06b78eee-0327-4d3e-b31a-b2adf904fc77">

First, it should display the following message (which is included in the response):
<img width="593" alt="Screenshot 2024-11-05 at 7 43 42 PM" src="https://github.com/user-attachments/assets/2e771e12-9667-4b2b-b305-a80971b6b8cf">

If the user tries to Validate, it will call a new API command `ResetPhoneNumberFailure`, which will return a `success` bool and a `message`. If unsuccessful:
<img width="558" alt="Screenshot 2024-11-05 at 8 02 12 PM" src="https://github.com/user-attachments/assets/e814b43b-a14a-49c8-97f6-65939c57e4a1">

If successful:
<img width="566" alt="Screenshot 2024-11-05 at 9 33 35 PM" src="https://github.com/user-attachments/assets/cab381f2-0d04-43bc-b4f6-48d119a12018">

If the user clicks "Send", then it should trigger this magic code and screen:
<img width="364" alt="Screenshot 2024-11-05 at 11 51 26 PM" src="https://github.com/user-attachments/assets/9d60e2e0-23e4-433e-9f23-a4e50e81c5e0">

Once this is done I'll do the integrated testing with the BE.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Create SMS delivery failure sign in flow in homepage
### What is the root cause of that problem?

This is a new feature request
### What changes do you think we should make in order to solve the problem?

1. We need to create an `SMSDeliveryModal` component like this design in the description of the OP. The initial is something like this. We can refactor the style in the PR phrase.

<img width="1512" alt="Screenshot 2024-11-06 at 11 52 27" src="https://github.com/user-attachments/assets/2693975b-fa71-448e-89bc-bc55b77780f1">


2. When we click the `Validate` button, we will call `ResetPhoneNumberFailure` API. 

- If `success` is `false`, we can change the content in `SMSDeliveryModal` to this case and the `message` is the `message` response in `ResetPhoneNumberFailure` API.

<img width="538" alt="Screenshot 2024-11-06 at 11 55 29" src="https://github.com/user-attachments/assets/0cefdff6-9988-4bfe-8820-5ea48712c8e5">

- If `success` is `true`, change the content in `SMSDeliveryModal` to this case, and when we click on `Send` button we will call `resendValidateCode` API and close the `SMSDeliveryModal` modal

<img width="537" alt="Screenshot 2024-11-06 at 11 58 45" src="https://github.com/user-attachments/assets/6d914866-d2b1-4439-86d9-d51bafc84f44">

3. To control the content in `SMSDeliveryModal`, we can use the data that is returned from `ResetPhoneNumberFailure` and a state to know `ResetPhoneNumberFailure` is called or not.

- Display the first screen if the `ResetPhoneNumberFailure` is not called. 
- Display the fail screen if `success` is `false`
- Display the success screen if `success` is `true`
 
4. Add this modal to the `SignInPage` here. In the `SignInPage`, we should create a new state `shouldShowSMSDelivery` with the initial value `true`. Then, when we click on the `Close` button, we will set it to `false`.

```
<SMSDeliveryModal
    isVisible={shouldShowSMSDelivery && !!account?.hasSMSDeliveryFailure}
    onClose={() => {
        setShouldShowSMSDelivery(false);
    }}
    login={userLogin}
/>
```
https://github.com/Expensify/App/blob/7bd88ac86ef786ac6d8c200f26331a02e9143eb6/src/pages/signin/SignInPage.tsx#L309
### What alternative solutions did you explore? (Optional)

**Comment by shawnborton** [comment]
I don't think we need to show a modal on NewDot, I think something more like @nyomanjyotisa is showing above makes the most sense.

**Comment by carlosmiceli** [comment]
Any updates @nyomanjyotisa ?

**Comment by carlosmiceli** [comment]
No, as I mentioned on the description, this needs to be tested internally with the BE once it's ready. The work here is to have it ready for testing, you can hardcode values to show that the logic works in a video.

**Comment by carlosmiceli** [comment]
@nyomanjyotisa question: how advanced are you with this PR? Because there's been a change of logic and displays that we've decided to implement recently. It should be less work actually, but I'm not sure how much you've done already.

@mallenexpensify how do we handle situations like this?

**Comment by carlosmiceli** [comment]
Great, I'll check out that branch and take it from here! Thank you!

@mallenexpensify @shawnborton Not sure how to handle this situation, I think Nyoman's job here is done, but I'll need to continue making PR changes based on some new decisions we made in for the BE. Can we consider this complete?

**Comment by carlosmiceli** [comment]
@mountiny @mallenexpensify @luacmartins need some advice here... We need to make some changes to the original issue's description, but Nyoman already completed the PR (for the original description).

Should this be considered done and we should merge and I work on a new PR (since I have to do the QA internally)?
Should Nyoman work on the new changes instead, and we should increase payment?
Should I work on top of Nyoman's PR somehow?

**Comment by luacmartins** [comment]
I agree with @parasharrajat. Let's see what the changes are first. I'd say if they are small, we should just update the current PR to reflect the changes. If they are larger (and the current PR won't introduce unwanted behavior), we can address them as a follow up in a new PR.

**Comment by carlosmiceli** [comment]
Sounds good. Need to get confirmation from one more potential change and I'll report back here with the final decision! Thanks guys!

**Comment by mountiny** [comment]
Yeah, lets just see the scope of the change but we can always just ask them to implement it and we can consider increasing the reward

**Comment by carlosmiceli** [comment]
Ok @nyomanjyotisa, I updated the description with the changes. To clarify, we are now making the flow shorter, by removing the previous Send screen if `hasSMSDeliveryFailure` returns `false`, and just showing the SMS failure time message if it's `true`. Let me know if you have any questions!

Please add a video simulating the error screen, and one completing the sign in flow into the app. I'll be back to review on Tuesday if it's ready, thank you!

**Comment by carlosmiceli** [comment]
@luacmartins @mountiny I think it's a pretty small change, but I'll defer to your judgement on how we should treat this change in the task.

**Comment by luacmartins** [comment]
It seems like these are small changes, so I'd just incorporate them in the existing PR.

**Comment by MitchExpensify** [comment]
Not overdue, Melvin. Actively being worked on

**Comment by carlosmiceli** [comment]
Sounds good @luacmartins, are you good to proceed @nyomanjyotisa ?

**Comment by mountiny** [comment]
@carlosmiceli yes, lets just ask @nyomanjyotisa to incorporate this change especially if its small

**Comment by carlosmiceli** [comment]
@nyomanjyotisa any updates on this? To clarify, this needs to have daily progress/updates, but your last [message](https://github.com/Expensify/App/issues/52086#issuecomment-2472038966) was a week ago.

**Comment by carlosmiceli** [comment]
Almost! This is the copy:

`We've been unable to deliver SMS messages to ${login}, so we've suspended it for 24 hours. {message}`

**Comment by carlosmiceli** [comment]
No, it has to be internally tested once this is merged. I'll take over that part, don't worry :)

**Comment by carlosmiceli** [comment]
I think this issue is going to need another PR...

Testing with expensify.com, this is what should be displayed if SMS status is failure:

![Screenshot 2024-11-28 at 9 03 52 PM](https://github.com/user-attachments/assets/30d988ad-771b-45b8-8eba-519d6bb17d22)

But same response structure in ND doesn't work and it actually resends the validation code. So we also need to fix it so we don't call Auth twice:

![Screenshot 2024-11-28 at 9 04 29 PM](https://github.com/user-attachments/assets/1b209089-110f-4e16-9b36-fa442dfd6a40)
![Screenshot 2024-11-28 at 9 04 44 PM](https://github.com/user-attachments/assets/244c0d45-5aaf-4024-b525-5478ed012560)

cc @nyomanjyotisa @parasharrajat can we put up a PR fix? Thanks!

**Comment by carlosmiceli** [comment]
Ah, that's a good catch! Yeah, let's do it lowercase instead, and can you send a video example of it working (hardcoding the api response)? Thank you!

**Comment by carlosmiceli** [comment]
Why does it show the validate code screen for a few moments instead of a loading animation?

**Comment by carlosmiceli** [comment]
Got it, that's great then! Thanks for the explanation :)

**Comment by carlosmiceli** [comment]
@nyomanjyotisa working almost great! can we get up a PR to fix this unnecessary ResendValidateCode auth call? It shouldn't be sent if it didn't pass the SMS check.

<img width="1154" alt="Screenshot 2024-12-02 at 4 32 15 PM" src="https://github.com/user-attachments/assets/a28be3d9-212b-42ba-8a01-10a21829fd7c">

**Comment by carlosmiceli** [comment]
Ohhhh, I see now... Thanks, I'll take care of it!

**Comment by carlosmiceli** [comment]
Not overdue Melv.

**Comment by MitchExpensify** [comment]
**Payment summary:**

- $250 @parasharrajat requires payment through NewDot Manual Requests
- $250  @nyomanjyotisa requires payment [automatic offer](https://www.upwork.com/nx/wm/offer/104794580) (Contributor)

**Comment by MitchExpensify** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by MitchExpensify** [comment]
Please complete the BZ steps above @parasharrajat , thanks

**Comment by MitchExpensify** [comment]
Ah yes. Ok all that is waiting is @parasharrajat's payment via NewDot, closing

**Comment by JmillsExpensify** [comment]
$250 approved for @parasharrajat
